rq/registry.py

```diff
@@ -268,6 +270,12 @@ def cleanup(self, timestamp: Optional[float] = None):
                         self.death_penalty_class, AbandonedJobError, AbandonedJobError(), traceback.extract_stack()
                     )
 
+                    # we did a job fetch so the meta is supposed to be there
+                    if job.meta:
+                        if ret := job.meta.get('_retried_after_abandonned', 0) < MAX_RETRIES_AFTER_ABANDONED:
+                            job.meta['_retried_after_abandonned'] = ret + 1
+                            self.retries_left = getattr(job, 'retries_left', 0) + 1
+
                     retry = job.retries_left and job.retries_left > 0
```

Comment:
Wait wait wait so ... if a job has some retries left, it will actually already be retried, right? (I thought that it would be failed)
So the "solution" of adding a retry strategy _actually_ fixes the problem, except if the job fails all his retries?

Reply:
technically yes, but I've been thinking about it and:
- I don't like that retrying because the job was "abandoned" consumes a "legitimate" retry, if you see what I mean.
- I don't like that we would (un *our* flask_rq or similar) add retries just to pass abandoned job situation.
I thought about leveraging the `execute_failure_callback` to retry the job if the error is AbandonedJob, but execute_failure_callback is already used to implement our dependancy mechanism so we would like to be extra cautious (current implementation is that there is only one callback, not a stack of callback)
Happy to discuss, and maybe you have a better idea ?

Reply:
I get your points. Philosophically speaking though, at the end of the day we'll be retrying the job (however we want to call it)

My take:
 - I agree we should distinguish between the two, and set a max of "abandoned retries"
 - I even think we should have some kind of alerts for jobs that were abandoned, and for which all retries were abandoned?